sd-validation/schema.js

```diff
@@ -0,0 +1,143 @@
+/**
```

Comment:
maybe rename to something more descriptive of what's happening in here, `schema-validator.js` or something?